include/nil/crypto3/hash/md5.hpp

```diff
@@ -40,8 +39,12 @@ namespace nil {
                         constexpr static const std::size_t digest_bits = policy_type::digest_bits;
                     };
 
+                    typedef detail::md_finalizer<typename params_type::digest_endian, block_cipher_type::word_bits,
+                                                 block_cipher_type::key_bits>
+                        md_finalizer;
```

Comment:
Not sure if the finalizer name has to be unique for each hash. Breaks concept-compliance.